### PR TITLE
Student Integrate Hiring

### PR DESCRIPTION
**Pull request by hoanghongminh**

### CODE REVIEW

File: src/app/event/[id]/page.tsx [hunk 1]
```diff
@@ -1,72 +1,91 @@
...
 
 export default function Page({ params }: { params: { id: string } }) {
-  const { eventDetail, updateEventDetail } = useEventDetail();
-
   const router = useRouter();
+  const { eventDetail } = useEventDetail(params.id);
+
+  const date = new Date();
+  const currentDay = date.getDay();
+  const dayNames = [
```
Comment (nxthai23) [inline src/app/event/[id]/page.tsx:14]: move to const file

File: src/app/event/[id]/page.tsx [hunk 2]
```diff
@@ -1,72 +1,91 @@
...
+import { format } from "date-fns";
 
 export default function Page({ params }: { params: { id: string } }) {
-  const { eventDetail, updateEventDetail } = useEventDetail();
-
   const router = useRouter();
+  const { eventDetail } = useEventDetail(params.id);
+
+  const date = new Date();
+  const currentDay = date.getDay();
```
Comment (nxthai23) [inline src/app/event/[id]/page.tsx:13]: why currentDay = new Date().getDay()? Should we get day from eventDetail.startTime/endTime here?

File: src/app/event/[id]/page.tsx [hunk 3]
```diff
@@ -1,72 +1,91 @@
...
-import { useRouter } from "next/router";
+import { useRouter } from "next/navigation";
 import { useEventDetail } from "@/lib/hooks/useEventDetail";
+import { format } from "date-fns";
 
 export default function Page({ params }: { params: { id: string } }) {
-  const { eventDetail, updateEventDetail } = useEventDetail();
-
   const router = useRouter();
+  const { eventDetail } = useEventDetail(params.id);
```
Comment (nxthai23) [inline src/app/event/[id]/page.tsx:10]: anyway to check eventDetail exist or not => render not found page?

File: src/app/event/[id]/page.tsx [hunk 4]
```diff
@@ -1,72 +1,91 @@
...
+    "Thứ hai",
+    "Thứ ba",
+    "Thứ tư",
+    "Thứ năm",
+    "Thứ sáu",
+    "Thứ bảy",
+  ];
+  const dayName = dayNames[currentDay];
+
+  const title = !eventDetail ? "" : eventDetail.attributes.title;
```
Comment (nxthai23) [inline src/app/event/[id]/page.tsx:25]: try using lodash here. Ex: title = get(eventDetail, 'attributes.title', '');

File: src/app/event/layout.tsx [hunk 5]
```diff
@@ -6,24 +6,33 @@ import { CreateProfileModal } from "@/components/home/CreateProfileModal";
...
   const [userName, setUserName] = useState("HENRY NGUYEN");
   const [userRank, setUserRank] = useState("BẠC IV");
   const [show, hide] = useProfileStore((state) => [state.show, state.hide]);
   const handleOpenModal = () => {
     show();
   };
+
+  const router = useRouter();
+  const handleClick = () => {
+    router.push("/home");
```
Comment (nxthai23) [inline src/app/event/layout.tsx:21]: make /home as const, then using const here.
Ex: ROUTES = {
HOME: '/home'
}
=> router.push(ROUTES.HOME);

File: src/app/home/layout.tsx [hunk 6]
```diff
@@ -25,20 +25,26 @@ export default function Layout({
   const handleOpenModal = () => {
     show();
   };
-  const events = useEvents().slice(1, 4);
+  const events = useEvents().slice(0, 3);
   const projects = useProjects().slice(1, 5);
 
+  const router = useRouter();
+  const handleClick = () => {
+    router.push("/home");
```
Comment (nxthai23) [inline src/app/home/layout.tsx:33]: check all layout using this hardcore

File: src/app/news/layout.tsx [hunk 7]
```diff
@@ -19,21 +19,35 @@ export default function Layout({
   const [userRank, setUserRank] = useState("BẠC IV");
   const [show, hide] = useProfileStore((state) => [state.show, state.hide]);
   const router = useRouter();
+  const pathname = usePathname();
+
+  const id = Number(pathname.split("/")[2]);
+  const isCustomLayout = !!id;
```
Comment (nxthai23) [inline src/app/news/layout.tsx:25]: why need this? add some comment here

File: src/components/recruitment/RecruitmentCard.tsx [hunk 8]
```diff
@@ -23,19 +24,20 @@ export const RecruitmentCard = ({
...
-            return (
-              <div
-                key={index}
-                className="px-2 py-1 bg-gray-200 text-xs text-gray-500 rounded-md"
-              >
-                {tag}
-              </div>
-            );
-          })}
+        {metadata &&
```
Comment (nxthai23) [inline src/components/recruitment/RecruitmentCard.tsx:31]: use lodash here

File: src/lib/hooks/useEvent.ts
Comment (nxthai23) [inline src/lib/hooks/useEvent.ts:1]: make meaningful name here, Ex: useFetchEvents()

File: src/lib/hooks/useEvent.ts [hunk 9]
```diff
@@ -1,20 +1,18 @@
...
 
   useEffect(() => {
     const fetchEvents = async () => {
       try {
-        const response = await axios.get(API_URL);
+        const response = await getEvent();
         setEvents(response.data);
       } catch (error) {
-        console.error("Error fetching event cards:", error);
+        console.error("Error fetching new cards:", error);
```
Comment (nxthai23) [inline src/lib/hooks/useEvent.ts:15]: throw error?

File: src/lib/hooks/useEventDetail.ts [hunk 10]
```diff
@@ -1,40 +1,38 @@
...
+export const useEventDetail = (paramsId: string) => {
+  const [eventDetail, setEventDetail] = useState<EventDetail | null>(null);
 
   useEffect(() => {
     const fetchEventDetail = async () => {
       try {
-        const response = await axios.get(API_URL);
-
-        setEventDetail(response.data);
+        const response = await tekdojoAxios.get<EventDetail>(
```
Comment (nxthai23) [inline src/lib/hooks/useEventDetail.ts:14]: move this to request file

File: src/lib/hooks/useNew.ts
Comment (nxthai23) [inline src/lib/hooks/useNew.ts:1]: fix same issue that mentioned above

File: src/lib/hooks/useNewDetail.ts
Comment (nxthai23) [inline src/lib/hooks/useNewDetail.ts:1]: same here

File: src/lib/hooks/useRecruitmentDetail.ts
Comment (nxthai23) [inline src/lib/hooks/useRecruitmentDetail.ts:1]: same here

File: src/lib/hooks/useRecruitmentDetail.ts [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+import { get } from "lodash";
+
+export const useRecruitmentDetail = (paramsId: string) => {
+  const [recruitmentDetail, setRecruitmentDetail] =
+    useState<RecruitmentDetail | null>(null);
+
+  useEffect(() => {
+    const fetchRecruitmentDetail = async () => {
+      try {
+        const response = await tekdojoAxios.get<RecruitmentDetail>(
```
Comment (nxthai23) [inline src/lib/hooks/useRecruitmentDetail.ts:15]: why dont use getHiring() from request.ts here?

**Review by nxthai23 — COMMENTED** [review]
review round 1
@hoanghongminh

File: src/app/event/layout.tsx (hunk 5 above)
Comment (hoanghongminh) [reply src/app/event/layout.tsx]: I fixed

File: src/app/home/layout.tsx (hunk 6 above)
Comment (hoanghongminh) [reply src/app/home/layout.tsx]: yes, I fixed => using const router

File: src/app/news/layout.tsx (hunk 7 above)
Comment (hoanghongminh) [reply src/app/news/layout.tsx:26]: I commented

**Pull request closed without merging**